internal/talos/talos_cluster_health_data_source.go

```diff
@@ -0,0 +1,201 @@
+				Description: "endpoints to use for the health check client. Use at least one control plane endpoint.",
+			},
+			"control_plane_nodes": schema.ListAttribute{
+				Required:    true,
+				ElementType: types.StringType,
+				Description: "List of control plane nodes to check for health.",
+			},
+			"worker_nodes": schema.ListAttribute{
+				Optional:    true,
+				ElementType: types.StringType,
+				Description: "List of worker nodes to check for health.",
+			},
+			"client_configuration": schema.SingleNestedAttribute{
+				Attributes: map[string]schema.Attribute{
+					"ca_certificate": schema.StringAttribute{
+						Required:    true,
+						Description: "The client CA certificate",
+					},
+					"client_certificate": schema.StringAttribute{
+						Required:    true,
+						Description: "The client certificate",
+					},
+					"client_key": schema.StringAttribute{
+						Required:    true,
+						Sensitive:   true,
+						Description: "The client key",
+					},
+				},
+				Required:    true,
+				Description: "The client configuration data",
+			},
+			"timeouts": timeouts.Attributes(ctx, timeouts.Opts{
+				Read: true,
+			}),
+		},
+	}
+}
+
+func (d *talosClusterHealthDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state talosClusterHealthDataSourceModelV0
+
+	diags := req.Config.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var (
+		endpoints         []string
+		controlPlaneNodes []string
+		workerNodes       []string
+	)
+
+	resp.Diagnostics.Append(state.Endpoints.ElementsAs(ctx, &endpoints, true)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(state.ControlPlaneNodes.ElementsAs(ctx, &controlPlaneNodes, true)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(state.WorkerNodes.ElementsAs(ctx, &workerNodes, true)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	readTimeout, diags := state.Timeouts.Read(ctx, 10*time.Minute)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	talosConfig, err := talosClientTFConfigToTalosClientConfig(
+		"dynamic",
+		state.ClientConfiguration.CA.ValueString(),
+		state.ClientConfiguration.Cert.ValueString(),
+		state.ClientConfiguration.Key.ValueString(),
+	)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to generate talos config", err.Error())
+
+		return
+	}
+
+	// since we're going to run this on a fully, this needs full PKI and all nodes/endpoins, so directly creating client
+	// and running the health check
+	c, err := client.New(ctx, client.WithConfig(talosConfig), client.WithEndpoints(endpoints...))
+	if err != nil {
+		resp.Diagnostics.AddError("failed to create talos client", err.Error())
+
+		return
+	}
+
+	defer c.Close() //nolint:errcheck
+
+	healthCheckClient, err := c.ClusterHealthCheck(ctx, readTimeout, &cluster.ClusterInfo{
```

Comment:
cool, using server-side health check API!

Reply:
I'm not sure why we have the client side now (we could drop it I guess? from talos)